local/include/core/http.h

```diff
@@ -58,6 +58,87 @@ static const swoc::Lexicon<UrlPart> URL_PART_NAMES{
      {UrlPart::Fragment, {URL_PART_FRAGMENT}}},
     {UrlPart::Error}};
 
+// This should be in the same order as "nghttp2_frame_type".
+enum class H2Frames {
+  DATA = 0x00,
+  HEADERS = 0x01,
+  PRIORITY = 0x02,
+  RST_STREAM = 0x03,
+  SETTINGS = 0x04,
+  PUSH_PROMISE = 0x05,
+  PING = 0x06,
+  GOAWAY = 0x07,
+  WINDOW_UPDATE = 0x08,
+  CONTINUATION = 0x09,
+  ALTSVC = 0x0a,
+  ORIGIN = 0x0c
+};
+
+static std::map<std::string, H2Frames, std::less<>> const h2_frames = {
+  {"DATA", H2Frames::DATA},
+  {"HEADERS", H2Frames::HEADERS},
+  {"PRIORITY", H2Frames::PRIORITY},
+  {"RST_STREAM", H2Frames::RST_STREAM},
+  {"SETTINGS", H2Frames::SETTINGS},
+  {"PUSH_PROMISE", H2Frames::PUSH_PROMISE},
+  {"PING", H2Frames::PING},
+  {"GOAWAY", H2Frames::GOAWAY},
+  {"WINDOW_UPDATE", H2Frames::WINDOW_UPDATE},
+  {"CONTINUATION", H2Frames::CONTINUATION},
+  {"ALTSVC", H2Frames::ALTSVC},
+  {"ORIGIN", H2Frames::ORIGIN}
+};
+
+static std::map<H2Frames, std::string> const h2_frame_name = {
+  {H2Frames::DATA, "DATA"},
+  {H2Frames::HEADERS, "HEADERS"},
+  {H2Frames::PRIORITY, "PRIORITY"},
+  {H2Frames::RST_STREAM, "RST_STREAM"},
+  {H2Frames::SETTINGS, "SETTINGS"},
+  {H2Frames::PUSH_PROMISE, "PUSH_PROMISE"},
+  {H2Frames::PING, "PING"},
+  {H2Frames::GOAWAY, "GOAWAY"},
+  {H2Frames::WINDOW_UPDATE, "WINDOW_UPDATE"},
+  {H2Frames::CONTINUATION, "CONTINUATION"},
+  {H2Frames::ALTSVC, "ALTSVC"},
+  {H2Frames::ORIGIN, "ORIGIN"}
+};
+
+// The status codes for the RST_STREAM and GOAWAY frames.
+enum class H2ErrorCodes {
+  NO_ERROR = 0x00,
+  PROTOCOL_ERROR = 0x01,
+  INTERNAL_ERROR = 0x02,
+  FLOW_CONTROL_ERROR = 0x03,
+  SETTINGS_TIMEOUT = 0x04,
+  STREAM_CLOSED = 0x05,
+  FRAME_SIZE_ERROR = 0x06,
+  REFUSED_STREAM = 0x07,
+  CANCEL = 0x08,
+  COMPRESSION_ERROR = 0x09,
+  CONNECT_ERROR = 0x0a,
+  ENHANCE_YOUR_CALM = 0x0b,
+  INADEQUATE_SECURITY = 0x0c,
+  HTTP_1_1_REQUIRED = 0x0d
+};
+
+static std::map<std::string, H2ErrorCodes, std::less<>> const h2_error_codes = {
+  {"NO_ERROR", H2ErrorCodes::NO_ERROR},
+  {"PROTOCOL_ERROR", H2ErrorCodes::PROTOCOL_ERROR},
+  {"INTERNAL_ERROR", H2ErrorCodes::INTERNAL_ERROR},
+  {"FLOW_CONTROL_ERROR", H2ErrorCodes::FLOW_CONTROL_ERROR},
+  {"SETTINGS_TIMEOUT", H2ErrorCodes::SETTINGS_TIMEOUT},
+  {"STREAM_CLOSED", H2ErrorCodes::STREAM_CLOSED},
+  {"FRAME_SIZE_ERROR", H2ErrorCodes::FRAME_SIZE_ERROR},
+  {"REFUSED_STREAM", H2ErrorCodes::REFUSED_STREAM},
+  {"CANCEL", H2ErrorCodes::CANCEL},
+  {"COMPRESSION_ERROR", H2ErrorCodes::COMPRESSION_ERROR},
+  {"CONNECT_ERROR", H2ErrorCodes::CONNECT_ERROR},
+  {"ENHANCE_YOUR_CALM", H2ErrorCodes::ENHANCE_YOUR_CALM},
+  {"INADEQUATE_SECURITY", H2ErrorCodes::INADEQUATE_SECURITY},
+  {"HTTP_1_1_REQUIRED", H2ErrorCodes::HTTP_1_1_REQUIRED}
+};
+
```

Comment:
It would be nice to also make this a Lexicon and allow the user to specify either the integer code or the string name for the abort code.